sympy/physics/mechanics/functions.py

```diff
@@ -532,7 +539,7 @@ def _recurser(expr, sub_dict):
 
 def _fraction_decomp(expr):
     """Return num, den such that expr = num/den"""
-    if not isinstance(expr, Mul):
+    if not isinstance(expr, MulClass):
```

Comment:
Couldn't Complex pretend to be an Add, for the purposes of its SymPy methods (like .args)? I suppose an Add with a complex would also need to pretend that it is a part of it (like `Add(1, I, x).args` should give `(1, I, x)`, not `(Complex(1+I), x)`).  

That's the sort of thing that the wrapper would have to do, fake the SymPy API everywhere. If the SymPy API is bad, say, because it forces a slowdown, we can discuss if it should be changed.